### PR TITLE
#35 Decouple intercepting from persisting

### PR DESCRIPTION
**Pull request by lukasz-gryzbon**
To meet the memory footprint requirements the [LMAX Disruptor](https://lmax-exchange.github.io/disruptor/) library was used.

### CODE REVIEW

File: src/main/resources/META-INF/spring-configuration-metadata.json [hunk 1]
```diff
@@ -3,6 +3,11 @@
     {
       "name": "lsd.dist.obfuscator.sensitiveHeaders",
       "type": "java.lang.String"
+    },
+    {
+      "name": "lsd.dist.queue.length",
```
Comment (nickmcdowall) [inline src/main/resources/META-INF/spring-configuration-metadata.json]: just wondering about the name of the property 🤔 , queue length threw me a bit - I had to double check that it was actually the pool size and not a limit on number of events or something like that.

File: src/main/java/io/lsdconsulting/lsd/distributed/interceptor/persistance/QueueService.kt [hunk 2]
```diff
@@ -0,0 +1,41 @@
+package io.lsdconsulting.lsd.distributed.interceptor.persistance
+
+import io.lsdconsulting.lsd.distributed.access.model.InterceptedInteraction
+import io.lsdconsulting.lsd.distributed.access.repository.InterceptedDocumentRepository
+import io.lsdconsulting.lsd.distributed.interceptor.config.log
+import java.util.concurrent.*
+import javax.annotation.PostConstruct
+import javax.annotation.PreDestroy
+
+class QueueService(
```
Comment (nickmcdowall) [inline src/main/java/io/lsdconsulting/lsd/distributed/interceptor/persistance/QueueService.kt]: minor thoughts: Just pondering on the name again - do we need to mention anything about queuing? Maybe that's an implementation detail that might change 🤔 . (Also my first reaction was to think of pub/sub queues when I saw the name 😆 ). 

Tricky to think of alternatives though.. would RepositoryService work better? 🤷

**Review by nickmcdowall — APPROVED** [review]
Hey looking good. Nice and neat solution. 😄 👍 

Only constructive feedback I could find is related to naming - couple minor comments added in case you want to consider.

File: src/main/java/io/lsdconsulting/lsd/distributed/interceptor/persistance/QueueService.kt (hunk 2 above)
Comment (lukasz-gryzbon) [reply src/main/java/io/lsdconsulting/lsd/distributed/interceptor/persistance/QueueService.kt]: Yes, I see your point. 

It was a proper in-memory pub/sub queue until yesterday 😄 

`RepositoryService` sounds good to me! Thanks!

File: src/main/resources/META-INF/spring-configuration-metadata.json (hunk 1 above)
Comment (lukasz-gryzbon) [reply src/main/resources/META-INF/spring-configuration-metadata.json]: I agree. Any suggestions?

**Comment by lukasz-gryzbon** [comment]
> To meet the memory footprint requirements the [LMAX Disruptor](https://lmax-exchange.github.io/disruptor/) library was used.

This was replaced with a `ThreadPoolExecutor` eventually.

**Pull request merged**